src/OneExplorer/OneExplorer.ts

```diff
@@ -150,17 +150,30 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
   private fileWatcher =
       vscode.workspace.createFileSystemWatcher(`**/*.{cfg,tflite,onnx,circle,tvn}`);
 
+  private tree: Node|undefined;
+
   constructor(private workspaceRoot: vscode.Uri) {
     const fileWatchersEvents =
         [this.fileWatcher.onDidCreate, this.fileWatcher.onDidChange, this.fileWatcher.onDidDelete];
 
     for (let event of fileWatchersEvents) {
-      event(() => this._onDidChangeTreeData.fire());
+      event(() => this.refresh());
     }
   }
 
-  refresh(): void {
-    this._onDidChangeTreeData.fire();
+  /**
+   * Refresh the tree under the given oneNode
+   * @param oneNode A start node to rebuild. The sub-tree under the node will be rebuilt.
+   *                If not given, the whole tree will be rebuilt.
+   */
+  refresh(oneNode: OneNode|void): void {
+    if (!oneNode) {
+      // refresh the root
+      this.tree = undefined;
```

Comment:
Is it right?
- oneNode -> undefined
- tree -> undefined
- oneNode -> undefeind

Is it your intention?

Reply:
ref: https://github.com/Microsoft/TypeScript/issues/25481#issuecomment-403082159

how about `refresh(oneNode?: OneNode): void {` ?

Reply:
> Is it your intention?

Yes :-D

Reply:
IHMO, if so,

```ts
refresh(oneNode: OneNode|undefined): void {
// ...
if (oneNode === undefined) { // if you want to clarify more 
  this.tree = undefined;
  // and I don't think it's necessary that oneNode = undefined;
}
```

But, it's optional as always. 😃